webrtc.html

```diff
@@ -2075,6 +1826,314 @@
+
+            <p>When the <dfn id="dom-peerconnection-addtrack"><code title=
+            "">addTrack()</code></dfn> method is invoked, the user agent MUST
+            run the following steps:</p>
+
+            <ol>
+              <li>
+                <p>Let <var>connection</var> be the
+                <code><a>RTCPeerConnection</a></code> object on which the
+                <code><a>MediaStreamTrack</a></code>, <var>track</var>, is to be
+                added.</p>
+              </li>
+
+              <li>
+                <p>If <var>connection</var>'s <a href=
+                "#dom-peerconnection-signaling-state"><code>RTCPeerConnection</code>
+                signalingState</a> is <code>closed</code>, throw an
+                <code>InvalidStateError</code> exception and abort these
+                steps.</p>
+              </li>
+
+              <li>
+                <p>If an <code>RTCRtpSender</code> for <var>track</var> already exists in
+                <var>connection</var>'s <a href="#senders-set">set of senders</a>,
+                then abort these steps.</p>
+              </li>
+
+              <li>
+                <p>Create a new <code>RTCRtpSender</code> for <var>track</var>, add it to
+                 <var>connection</var>'s <a href="#senders-set">set of senders</a>,
+                 and return it to the caller.</p>
+              </li>
+
+              <li>
+                <p>A track could have contents that are inaccessible to the
+                application. This can be due to being marked with a
+                <var>peerIdentity</var> option or anything that would make a
+                track <a href=
+                "http://www.w3.org/html/wg/drafts/html/master/infrastructure.html#cors-cross-origin">
+                CORS cross-origin</a>. These tracks can be supplied to the
+                addTrack method, and have an <code>RTCRtpSender</code> created for them, but content
+                MUST NOT be transmitted, though tracks marked with
+                <var>peerIdentity</var> can be transmitted if they meet the
+                requirements for sending (see <a href="#isolated-pc">
+                isolated streams and RTCPeerConnection</a>).</p>
+
+                <p>All other tracks that are not accessible to the application
+                MUST NOT be sent to the peer, with silence (audio), black
+                frames (video) or equivalently absent content being sent in
+                place of track content.</p>
+
+                <p>Note that this property can change over time.</p>
+              </li>
+
+              <li>
+                <p>If <var>connection</var>'s <a href=
+                "#dom-peerconnection-signaling-state"><code>RTCPeerConnection</code>
+                signalingState</a> is <code>stable</code>, then fire a <a href=
+                "#event-negotiation">negotiationneeded</a> event at
+                <var>connection</var>.</p>
```

Comment:
The last two steps should be moved out of the algorithm. But let's do that in a separate PR.